bindings/node.js/kzg.cxx

```diff
@@ -152,190 +152,215 @@ Napi::Value BlobToKzgCommitment(const Napi::CallbackInfo& info) {
-    ).ThrowAsJavaScriptException();
+    Napi::TypeError::New(env, "Error in verifyBlobKzgProof").ThrowAsJavaScriptException();
     return env.Null();
   }
 
-  return Napi::Boolean::New(env, verification_result);
+  return Napi::Boolean::New(env, out);
 }
 
-// verifyKzgProof: (commitmentBytes: Bytes48, zBytes: Bytes32, yBytes: Bytes32, proofBytes: Bytes48, setupHandle: SetupHandle) => boolean;
-Napi::Value VerifyKzgProof(const Napi::CallbackInfo& info) {
+// verifyBlobKzgProofBatch: (blobs: Blob[], commitmentsBytes: Bytes48[], proofsBytes: Bytes48[], setupHandle: SetupHandle) => boolean;
+Napi::Value VerifyBlobKzgProofBatch(const Napi::CallbackInfo& info) {
   auto env = info.Env();
 
   size_t argument_count = info.Length();
-  size_t expected_argument_count = 5;
+  size_t expected_argument_count = 4;
   if (argument_count != expected_argument_count) {
     return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
   }
 
-  auto commitment_bytes = extract_byte_array_from_param(info, 0, "commitmentBytes");
-  auto z_bytes = extract_byte_array_from_param(info, 1, "zBytes");
-  auto y_bytes = extract_byte_array_from_param(info, 2, "yBytes");
-  auto proof_bytes = extract_byte_array_from_param(info, 3, "proofBytes");
-  auto kzg_settings = info[4].As<Napi::External<KZGSettings>>().Data();
+  auto blobs_param = info[0].As<Napi::Array>();
+  auto commitments_param = info[1].As<Napi::Array>();
+  auto proofs_param = info[2].As<Napi::Array>();
+  auto kzg_settings = info[3].As<Napi::External<KZGSettings>>().Data();
 
-  if (env.IsExceptionPending()) {
+  auto blobs_count = blobs_param.Length();
+  auto commitments_count = commitments_param.Length();
+  auto proofs_count = proofs_param.Length();
+
+  if (blobs_count != commitments_count || blobs_count != proofs_count) {
+    Napi::Error::New(env, "verifyBlobKzgProofBatch requires equal number of blobs/commitments/proofs")
+      .ThrowAsJavaScriptException();
     return env.Null();
   }
 
+  auto blobs = (Blob*)calloc(blobs_count, sizeof(Blob));
+   if (blobs == NULL) {
+     Napi::Error::New(env, "Error while allocating memory for blobs").ThrowAsJavaScriptException();
+     return env.Null();
+   };
+
+   auto commitments = (Bytes48*)calloc(commitments_count, sizeof(Bytes48));
+   if (commitments == NULL) {
+     free(blobs);
+     Napi::Error::New(env, "Error while allocating memory for commitments").ThrowAsJavaScriptException();
+     return env.Null();
+   };
+
+   auto proofs = (Bytes48*)calloc(proofs_count, sizeof(Bytes48));
+   if (proofs == NULL) {
+     free(blobs);
+     free(commitments);
+     Napi::Error::New(env, "Error while allocating memory for proofs").ThrowAsJavaScriptException();
+     return env.Null();
+   };
+
+  for (uint32_t index = 0; index < blobs_count; index++) {
+    // Extract blob bytes from parameter
+    Napi::Value blob = blobs_param[index];
+    auto blob_bytes = blob.As<Napi::Uint8Array>().Data();
```

Comment:
Do we need to do any checks here similar to what we are doing in `extract_byte_array_from_param()`?
I wonder if we can abstract this entire logic into a function instead of doing it three times explicitly (once for proofs, once for blobs, once for commitments).

Reply:
Let's look into this more later.